server/test/models/ApplicationModelTest.java

```diff
@@ -21,67 +19,13 @@ public class ApplicationModelTest extends ResetPostgres {
           .setStatusText("Approved")
           .setLocalizedStatusText(LocalizedStrings.withDefaultValue("Approved"))
           .build();
-  ApplicationStatusesRepository applicationStatusesRepository;
+  private ApplicationStatusesRepository applicationStatusesRepository;
+  private ApplicationEventRepository applicationEventRepository;
 
   @Before
   public void setUp() {
     applicationStatusesRepository = instanceOf(ApplicationStatusesRepository.class);
-  }
-
-  @Test
-  public void staleLatestStatusIsNotPersisted() {
-    // Tests a case where an Application (and its associated latest_status value has been loaded
-    // in-memory, a new ApplicationEventDetails is added (causing the trigger to execute), and the
-    // Application is persisted.
-    ProgramModel program = ProgramBuilder.newActiveProgram("test program", "description").build();
```

Comment:
**suggestion**: I think we actually do need to handle this race condition. Anywhere we update a submitted application, I think we're going to need to check if the `latest_status` field in the database matches the latest_status field on the `ApplicationModel` we're about to save, and if it doesn't, carry it forward to the newly updated application in the same transaction.